packages/shared/src/components/auth/OnboardingRegistrationForm4d5.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { ReactElement, useContext, useEffect, useState } from 'react';
+import { useMutation } from '@tanstack/react-query';
+import classNames from 'classnames';
+import { checkKratosEmail } from '../../lib/kratos';
+import { AuthFormProps, getFormEmail, providerMap } from './common';
+import OrDivider from './OrDivider';
+import AnalyticsContext from '../../contexts/AnalyticsContext';
+import { AuthEventNames, AuthTriggersType } from '../../lib/auth';
+import { Button, ButtonVariant } from '../buttons/ButtonV2';
+import AuthForm from './AuthForm';
+import { TextField } from '../fields/TextField';
+import { MailIcon } from '../icons';
+import { IconSize } from '../Icon';
+import Alert, { AlertParagraph, AlertType } from '../widgets/Alert';
+import { OnboardingHeadline } from './OnboardingHeadline';
+import AuthModalFooter from './AuthModalFooter';
+import Logo, { LogoPosition } from '../Logo';
+import { ModalClose } from '../modals/common/ModalClose';
+import { CloseAuthModalFunc } from '../../hooks/useAuthForms';
+
+const signupProviders = [providerMap.google, providerMap.github];
+
+interface OnboardingRegistrationForm4d5Props extends AuthFormProps {
+  onExistingEmail?: (email: string) => unknown;
+  onSignup?: (email: string) => unknown;
+  onProviderClick?: (provider: string, login?: boolean) => unknown;
+  targetId?: string;
+  isLoginFlow?: boolean;
+  logInTitle?: string;
+  signUpTitle?: string;
+  trigger: AuthTriggersType;
+  isReady: boolean;
+  className?: string;
+  onClose?: CloseAuthModalFunc;
+}
+
+export const OnboardingRegistrationForm4d5 = ({
+  onSignup,
+  onExistingEmail,
+  onProviderClick,
+  targetId,
+  isReady,
+  trigger,
+  className,
+  onClose,
+}: OnboardingRegistrationForm4d5Props): ReactElement => {
+  const { trackEvent } = useContext(AnalyticsContext);
+  const [shouldLogin, setShouldLogin] = useState(false);
+  const [registerEmail, setRegisterEmail] = useState<string>(null);
+  const { mutateAsync: checkEmail, isLoading } = useMutation(
+    (emailParam: string) => checkKratosEmail(emailParam),
+  );
+
+  useEffect(() => {
+    trackEvent({
+      event_name: shouldLogin
+        ? AuthEventNames.OpenLogin
+        : AuthEventNames.OpenSignup,
+      extra: JSON.stringify({ trigger }),
+      target_id: targetId,
+    });
+    // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Any issues with adding all dependencies here? If not, let's add and remove this 🙏

Reply:
yeah, our trackEvent function is not stable and we only wanted this event when login state changes so if we change it here it would change how many of events it sends thus impacting analytics.

Reply:
We can always compare the previous version of login state to the current and if changed we can track within an if condition while having all deps in place. 
But anyways should be fine, we already have a ton of these disables anyways.

Reply:
I agree, If you ask me this should just have all the dependencies inside but it was requested like this before from analytics. 👍